src/Compilers/CSharp/Portable/Binder/Binder_Lookup.cs

```diff
@@ -1058,11 +1058,21 @@ static void addAllInterfaces(NamedTypeSymbol @interface, HashSet<NamedTypeSymbol
             if (interfaces.Length > 0)
             {
                 var tmp = LookupResult.GetInstance();
-                foreach (TypeSymbol baseInterface in interfaces)
+                HashSet<NamedTypeSymbol> seenInterfaces = null;
```

Comment:
>HashSet [](start = 16, length = 7)

Would it make sense to use pooled hash set?

Reply:
The reason I didn't use the pool is that we'd need a new one for this specific comparer. Not sure that's warranted

---
In reply to: [544338611](https://github.com/dotnet/roslyn/pull/49338#discussion_r544338611) [](ancestors = 544338611)